wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/SyntaxRules.kt

```diff
@@ -0,0 +1,26 @@
+package com.squareup.wire.schema
+
+import com.squareup.wire.schema.ProtoFile.Syntax
+import com.squareup.wire.schema.ProtoFile.Syntax.PROTO_2
+import com.squareup.wire.schema.ProtoFile.Syntax.PROTO_3
+
+/** A set of rules which defines schema requirements for a specific [Syntax]. */
+interface SyntaxRules {
```

Comment:
internal

Reply:
Seems like we can't use `internal` interface as an argument in a public function. Let's fix this in a future follow-up.
```
e: wire/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/EnclosingType.kt: (32, 41): 'public' function exposes its 'internal' parameter type SyntaxRules
e: wire/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/EnumType.kt: (53, 41): 'public' function exposes its 'internal' parameter type SyntaxRules
e: wire/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/MessageType.kt: (134, 41): 'public' function exposes its 'internal' parameter type SyntaxRules
e: wire/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Type.kt: (33, 41): 'public' function exposes its 'internal' parameter type SyntaxRules
```